Dockerfile

```diff
@@ -3,23 +3,21 @@ FROM node:10
 # Create app directory
 WORKDIR /usr/src/app
 
-# copy output
+# First copy over the just the package.json files
+# so we can build a cached base image that only has node_modules
+# use the 'npm ci' command to get reproducable builds
+COPY package.json package-lock.json ./
+COPY server/package.json server/package-lock.json server/
+RUN npm ci && npm --prefix server ci 
+
+# Copy Everything Else
 COPY . .
 
-# Install npm packages
-# TODO use --only=production flag
-#
-RUN npm install
-
+# Build and test everything
 RUN ./tools/sigh && npm run build:rollup
+RUN npm --prefix server test
 
-WORKDIR /usr/src/app/server
-
-# This odd npm syntax solves a problem installing with an empty
-# node_modules directory: See https://npm.community/t/518
-RUN npm install --no-package-lock && npm install --package-lock-only
-
-RUN npm test
+#RUN npm install --no-package-lock && npm install --package-lock-only
```

Comment:
If we keep this commented out perhaps we should keep the comment that was here explaining, and just note that we're checking to see if this is no longer needed, and once confirmed we can remove it wholly.

Reply:
actually can just remove this, this is included in the new npm 'server' script.